src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1151,7 +1198,7 @@ object LazyList extends SeqFactory[LazyList] {
       *  @param hd   The first element of the result lazy list
       *  @param tl   The remaining elements of the result lazy list
       */
-    def apply[A](hd: => A, tl: => LazyList[A]): LazyList[A] = newLL(sCons(hd, newLL(tl.state)))
+    def apply[A](hd: => A, tl: => LazyList[A]): LazyList[A] = newLL(eagerCons(hd, newLL(tl)))
```

Comment:
both here and in `Deferrer`, you could pass `null` as the tail, which would yield an (incorrectly!) empty `LazyList`, rather than an exception. to this end, I propose we use a marker object instead of `null` as the value of `_state` (why isn't it called `_tail` btw?) to indicate emptiness. with the marker, a `null` tail would just be a programming error like it would be with `List` and wouldn't need any special handling IMO. the alternative as I see it is to add a whole bunch of `null` checks for the tail, which I think is less pleasant

Reply:
Thank you! I pushed a slightly different, simpler fix, please take a look.

The problem is that `EmptyFun: () => LazyList[A]` has to return a `LazyList`, so it cannot be an arbitrary marker object, it has to be a `LazyList` (or `null`, which is what we use).

Or we would change the type of `lazyState` to `() => AnyRef`, that should also work.

Reply:
I think with this change `LazyList.cons(null, null)` still reads as empty rather than `null` and then NPE

Reply:
I probably didn't get what you mean then, https://github.com/scala/scala/pull/10937/commits/188f445436ff55ace8b685dd4399e1a117b2d555#diff-b64326ffeb79c30e7645bbfd8c8071cd453e417debf25dfb870af8323ebeddf6 passes and is the same as in 2.13.15.

Reply:
you're right, I was mistaken. I forgot that `LazyList.cons(null, null)` translates to `newLL(eagerCons(null, (() => null)())`, not `newLL(eagerCons(null, null))`. also, I'm sorry for taking a week to get to this